apps/wizard/app_pages/map_brackets.py

```diff
@@ -0,0 +1,923 @@
+            reload_rank=True,
+            reload_optimal_brackets=True,
+        )
+    except Exception as e:
+        log.error(e)
+        st.error(e)
+        st.stop()
+
+    if mb.min_value < -mb.smallest_number:
+        st.warning("The optimal bracket search is only properly implemented for purely positive brackets.")
+
+    # Add toggles to control whether lower and upper brackets should be open.
+    _message = ""
+    if mb.latest_year:
+        _message = "Note that, even if set to close, a bracket may still remain open."
+        _message = " This happens when closing the bracket implies breaking the scale. For example, if the maximum data value is 100 and the scale is [0, 10, 20, 30, 40], closing the upper bracket would mean breaking the scale: [0, 10, 20, 30, 100], which is undesirable."
+        if mb.latest_year:
+            _message += " Also, it is possible that a value in a previous year exceeds the current brackets. For example, suppose the maximum data value for the latest year is 35, and brackets are [0, 10, 20, 30, 40]. The maximum bracket will still remain open if the maximum data value in a previous year is larger than 40."
+    mb.lower_bracket_open = st.toggle("Lower bracket open", mb.lower_bracket_open, help=_message)
+    mb.upper_bracket_open = st.toggle("Upper bracket open", mb.upper_bracket_open, help=_message)
+    mb.run(
+        reload_data_values=False,
+        reload_openness=False,
+        reload_all_brackets=True,
+        reload_rank=True,
+        reload_optimal_brackets=True,
+    )
+    mb.percentile_min, mb.percentile_max = st.slider(
+        label="Percentile range covered by brackets",
+        min_value=0,
+        max_value=100,
+        value=(MIN_PERCENTILE, MAX_PERCENTILE),
+        help="Currently, this only affects linear brackets",
+    )
+    mb.run(
+        reload_data_values=False,
+        reload_openness=False,
+        reload_all_brackets=True,
+        reload_rank=True,
+        reload_optimal_brackets=True,
+    )
+    if mb.percentiles_ignored:
+        st.warning(
+            "Minimum and maximum percentiles are identical. Percentiles are ignored. Consider broadening the percentile range."
+        )
+
+    # Select bracket type.
+    bracket_type_labels = {
+        # Add optimal choice
+        f"Optimal ({mb.brackets_optimal['optimal']})": mb.brackets_optimal["optimal"],
+        # Rest of the options
+        **{
+            value: value
+            for value in list(BRACKET_LABELS["log"].values())
+            + list(BRACKET_LABELS["linear"].values())
+            + list(BRACKET_LABELS["custom"].values())
+        },
+    }
+    # Create a bracket type selector with a select box.
+    bracket_type = st.selectbox(
+        "Select linear or log-like",
+        options=bracket_type_labels,
+        # NOTE: By default, the first option will be selected, which is expected to be the optimal one.
+        index=0,
+        # horizontal=True,
```

Comment:
help message explaining what this is? e.g. 'at owid we use different strategies for the map bracket bins. These are mostly divided into two categories: log and linear. ...'